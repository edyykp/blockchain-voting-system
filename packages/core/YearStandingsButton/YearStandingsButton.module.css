.button {
  width: 100%;
  margin-bottom: 20px;
  font-size: 1.5rem;
  border: 1px solid green;
  background-color: green;
  color: var(--gray-200);
  padding: 5px;
  border-radius: 5px;
  cursor: pointer;
  transition: 0.2s ease-in-out;
  box-shadow: 2px 2px 10px 1px green;
}

.button:hover {
  border: 1px solid var(--white);
  color: green;
  background-color: transparent;
}
