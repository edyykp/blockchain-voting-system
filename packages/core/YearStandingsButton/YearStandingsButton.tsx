import styles from './YearStandingsButton.module.css';
import { useSiteProperties } from '../../config/SiteProperties/useSiteProperties';

type YearStandingsButtonProps = {
  year: string;
};

export const YearStandingsButton = ({ year }: YearStandingsButtonProps) => {
  const valueOf = useSiteProperties();

  const text = {
    buttonText: valueOf('standings_button_text'),
  };

  return (
    <button className={styles.button} data-testid="year-standings-button">
      {text.buttonText} {year}
    </button>
  );
};
