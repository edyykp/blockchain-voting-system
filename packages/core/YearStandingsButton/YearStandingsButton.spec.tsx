import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { YearStandingsButton } from './YearStandingsButton';

const setup = (year: string) => {
  return {
    ...render(<YearStandingsButton year={year} />),
  };
};

describe('YearStandingsButtonâ†’', () => {
  it('renders', () => {
    const { container, getByTestId } = setup('2017');

    expect(getByTestId('year-standings-button')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders with correct year', () => {
    const { getByText } = setup('2022');

    expect(getByText('Standings 2022')).toBeInTheDocument();
  });
});
