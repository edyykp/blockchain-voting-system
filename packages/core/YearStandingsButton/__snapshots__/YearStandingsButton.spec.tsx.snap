// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearStandingsButtonâ†’ renders 1`] = `
<div>
  <button
    class="button"
    data-testid="year-standings-button"
  >
    Standings
     
    2017
  </button>
</div>
`;
